Sources/SwiftDocC/Infrastructure/DocumentationContext.swift

```diff
@@ -2398,6 +2400,37 @@ public class DocumentationContext: DocumentationContextDataProviderDelegate {
         }
         return automaticallyCuratedSymbols
     }
+
+    /// Handles overloaded symbols by grouping them together into one page.
+    private func groupOverloadedSymbols(with linkResolver: PathHierarchyBasedLinkResolver) throws {
+        guard FeatureFlags.current.isExperimentalOverloadedSymbolPresentationEnabled else {
+            return
+        }
+        
+        try linkResolver.traverseOverloadedSymbols { overloadedSymbolReferences in
+
+            // Tell each symbol what other symbols overload it.
+            for (index, symbolReference) in overloadedSymbolReferences.indexed() {
+                let documentationNode = try entity(with: symbolReference)
+
+                guard let symbol = documentationNode.semantic as? Symbol else {
+                    preconditionFailure("""
+                    Only symbols can be overloads. Found non-symbol overload for \(symbolReference.absoluteString.singleQuoted).
+                    Non-symbols should already have been filtered out in `PathHierarchyBasedLinkResolver.traverseOverloadedSymbols(_:)`.
+                    """)
+                }
+                guard symbolReference.sourceLanguage == .swift else {
```

Comment:
The current behavior, based on a quick test with this code, tells me the following:

- The `@objc` method correctly appears on the Objective-C view of `MyClass`.
- Selecting the `-> Int` version of the method in Swift enables the language picker on that page, as well as showing the overloads listing.
- Selecting the `-> String` version of the method disables the language picker, because there is no corresponding Objective-C method to switch to.

So i think it behaves correctly, from what i can tell. When the automatic curation tweaks are added so that only one page is linked in the curation hierarchy, we'll need to make sure that the individual overload selections still allow you to switch to the appropriate language view, but that's later work.